add verifyClient option for mock-socket Server constructor

jwoodrow opened the pull request:
would address https://github.com/romgain/jest-websocket-mock/issues/49

I've followed the contributing readme and there doesn't appear to be any breaking changes from what I've submitted.

if there are cleaner tests or better tests you would suggest I'll try and do what I can

romgain commented:
Thank you for the contribution!

This is a good idea, thanks!
I think it would also be useful to handle the other mock-socket options. Would you mind tweaking your implementation to just forward options other than jsonProtocol to the mock-socket constructor?

In that case, could you also add unit tests for the other options supported by mock-socket, and documenting it in the README?

Thanks again 👍

jwoodrow commented:
I've looked at the `mock-socket` server/client implementation and I only see `verifyClient` and `selectProtocol` as used options AFAIK so I've added `selectProtocol` to the PR with tests and also cleaned up my code (it's not my first time using TS but my second so I'm finding better ways to define types as I go 😅)

I've also updated the readme.

romgain commented on src/websocket.ts:
```diff
@@ -36,9 +39,10 @@ export default class WS {
+  constructor(url: string, opts: WSOptions = {}) {
     WS.instances.push(this);
 
+    const { jsonProtocol, ...serverOptions } = opts;
```
This should be something like:

```suggestion
    const { jsonProtocol = false, ...serverOptions } = opts;
```

romgain commented on src/websocket.ts:
```diff
@@ -3,8 +3,11 @@ import Queue from "./queue";
 interface WSOptions {
   jsonProtocol?: boolean;
+  verifyClient?: () => boolean;
+  selectProtocol?: (protocols: string[]) => string;
```
Can you import `ServerOptions` from mock-socket and just have this interface extend it to add the `jsonProtocol` option? 
https://github.com/thoov/mock-socket/blob/master/index.d.ts#L88

romgain reviewed (approved):
Thanks @jwoodrow! I just tweaked the tests & readme a little, but it's good to go!
Thanks again :+1:
